lib/class-wp-rest-url-details-controller.php

```diff
@@ -320,11 +315,21 @@ private function get_image( $meta_elements, $url ) {
 		return $image;
 	}
 
+	/**
+	 * Convert HTML entities into HTML for use downstream.
+	 *
+	 * @param string $html The HTML to convert.
+	 * @return string The HTML with all entities converted.
+	 */
+	private function convert_html_entities( $html ) {
+		return html_entity_decode( $html, ENT_QUOTES, get_bloginfo( 'charset' ) );
```

Comment:
My only concern here is that we're actually allowing tags to be sent to the client here. I wonder whether we use [`wp_strip_all_tags`](https://developer.wordpress.org/reference/functions/wp_strip_all_tags/) to leave only the content, or make that some kind of opt-_in_ option via a filter.

Reply:
Why strip the tags? Is there a use case where the tags are or could be problematic? Or does the downstream not expect HTML, i.e. it expects a string?

Reply:
So thinking about this further I'd prefer to be safe and

1. Convert all entities to HTML form (✅)
2. Strip all HTML tags.

In the future, if folks find they _need_ HTML in the response we can always add a `raw` field and allow them to process on their end as a consumer of the API.

Reply:
In separate discussion, @getdave and I discussed and agreed to the following:

- There's no current or anticipated need for HTML within the metadata for downstream
- Having HTML in the metadata means consumers of the API will need to add escaping at the point of rendering

The plan is to design the API to output non-HTML metadata. Doing so ensures the content being rendered downstream is safe for users.

How?

- Strip all HTML tags and tag entities
- Convert any non-tags into their characters